### PR TITLE
feat(components/forms/enumeration): add tooltip (+delay), remove blur

### PR DESCRIPTION
**Pull request by Sylvain-Talend**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
- No tooltip on enumeration list of values
- When long name in list of values, no ellipsis
- No test for the enumeration widget
- Blur activated in list of values, meaning that even if we abort the edition, the value is saved!

**What is the new behavior?**
- tooltip on enumeration list of values
- Ellipsis when long name
- Tests added for the enumeration widget
- Blur desactivated on list of values

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

Important: 
- Add mock for snapshot testing (as tooltip generates id)
- The delay of the tooltip is now of 400 ms, FOR ALL COMPONENTS including the list. Please tell if it is not the wanted behavior for your case.

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

**Comment by jmfrancois** [comment]
tooltip should be "top" ?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

**Comment by Sylvain-Talend** [comment]
@jmfrancois Indeed the tooltip should be top.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

File: packages/components/src/Enumeration/Items/Item/Item.component.js [hunk 1]
```diff
@@ -61,12 +62,16 @@ function Item({ id, item }) {
...
-				className={itemLabelClasses()}
-			/>
+				tooltipPlacement="top"
+			>
+				<Button
+					key={item.index}
+					onClick={event => onSelectItem(item, event)}
+					className={itemLabelClasses()}
+				>{item[key].join(',')}</Button>
+			</TooltipTrigger>
```
Comment (jmfrancois) [inline packages/components/src/Enumeration/Items/Item/Item.component.js]: I would prefer to keep Action component here. 
So please add a props to always render the tooltip: 

props.tooltip with PropTypes.boolean

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

File: packages/components/src/Enumeration/Items/Item/Item.component.js (hunk 1 above)
Comment (Sylvain-Talend) [inline packages/components/src/Enumeration/Items/Item/Item.component.js]: Props added

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://277.talend.surge.sh)

**Pull request merged**